File: 27.remove_element.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        # Use 2 ids, left and right
        rid = len(nums)-1
        lid = rid

        while lid >= 0:
            if nums[rid] == val:
                rid -= 1
            elif nums[lid] == val:
                nums[lid] = nums[rid]
                rid -= 1
            lid-=1
        
        return rid+1

    def removeElement_optimized(self, nums: List[int], val: int) -> int:
        # Use 2 ids, left and right
        lid = 0

        for rid in range(len(nums)):
            if nums[rid] != val:
                nums[lid] = nums[rid]
                lid += 1
        
        return lid+1

File: 27.remove_element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int n = nums.size()-1, i = n;

        while(i>=0){
            if(nums[i] == val){
                nums[i] = nums[n];
                n--;
            }
            i--;
        }
        return n+1;
    }
};